lib/ecstatic/show-dir/index.js

```diff
@@ -108,15 +108,27 @@ module.exports = (opts) => {
             const iconClass = `icon-${isDir ? '_blank' : classForNonDir}`;
 
             // TODO: use stylessheets?
-            html += `${'<tr>' +
-              '<td><i class="icon '}${iconClass}"></i></td>`;
-            if (!hidePermissions) {
-              html += `<td class="perms"><code>(${permsToString(file[1])})</code></td>`;
+            html += '<tr>';
+            const selectDisplayOptions = selectDisplay.split('|');
+            for (let i = 0; i < selectDisplayOptions.length; i += 1) {
+              switch (selectDisplayOptions[i].trim()) {
+                case 'icon':
+                  html += `${'<td><i class="icon '}${iconClass}"></i></td>`;
```

Comment:
This approach means we can use display options to change the *order* in which elements are displayed. I think we can fix this by iterating over a hard-coded list order. `['icon', 'permissions', 'filesize', 'filename'].forEach(function(option) {`

Reply:
Yes. I meant it. 
But how to get an order since it should be an object? I believe the object does not guarantee its order.

Reply:
correct--afaik while in practice objects are sorted that the language spec doesn't include that property. So!

There are multiple ways to skin this, but maybe something like

```js
[
  ['hideIcons', `${'<td><i class="icon '}${iconClass}"></i></td>`],
  ['hidePermissions', `<td class="perms"><code>(${permsToString(file[1])})</code></td>`]
].forEach(function(pair) {
  if (opts[pair[0]]) {
    html += pair[1]
  }
});
```

A little gross maybe, and there's more than one way to skin this one for sure, but this is a strategy that might work.

Reply:
This reminds me of something though--I'll mark it in a separate comment.

Reply:
Well, what about templated html-page? I mean, in templated html we have strong order of html-elements. So, templated html-page could have something like this:
```handlebars
<table>
  {{#each files}}
    <tr>
      {{#if icon}}
        <td>
          <i class="icon {{iconType}}"></i>
        </td>
      {{/if}}
      {{#if permissions}}
        <td class="perms">
          <code>{{permissions}}</code>
        </td>
      {{/if}}
      {{#if filesize}}
        <td class="file-size">
            <code>{{size}}</code>
        </td>
      {{/if}}
      {{#if filename}}
        <td class="display-name">
          <a href={{href}}>{{name}}</a>
        </td>
      {{/if}}
    </tr>
  {{/each}}
</table>
```
But templated html page might be overkill for this task, so i don't know.